keep/api/core/dependencies.py

```diff
@@ -166,7 +185,21 @@ def verify_bearer_token(token: str = Depends(oauth2_scheme)) -> str:
                 leeway=60,
             )
             tenant_id = payload.get("keep_tenant_id")
-            return tenant_id
+            role_name = payload.get(
+                "keep_role", AdminRole.get_name()
+            )  # default to admin for backwards compatibility
+            email = payload.get("email")
+            role = get_role_by_role_name(role_name)
+            # validate scopes
+            if not role.has_scopes(scopes):
+                raise HTTPException(
+                    status_code=403,
+                    detail="You don't have the required permissions to access this resource",
+                )
+            return AuthenticatedEntity(tenant_id, email)
+        # authorization error
+        except HTTPException:
+            raise
```

Comment:
?

Reply:
HTTPExceptions are specific (e.g. 403 auth) which I don't want to be caught by the generic 401